docs/language/values-and-types.mdx

```diff
@@ -405,14 +405,26 @@ let aNumber = 0x436164656E636521
 // `aNumber` has type `Int`
 ```
 
-Address can also be created using a byte array.
+Address can also be created using a byte array or string.
 
 ```cadence
 // Declare an address with hex representation as 0x436164656E636521.
 let someAddress: Address = Address.fromBytes([67, 97, 100, 101, 110, 99, 101, 33])
 
 // Invalid: Provided value is not compatible with type `Address`. The function panics.
 let invalidAddress: Address = Address.fromBytes([12, 34, 56, 11, 22, 33, 44, 55, 66, 77, 88, 99, 111])
+
+// Declare an address with the string representation as "0x436164656E636521".
+let addressFromString: Address = Address.fromString("0x436164656E636521")
+
+// Declare an address with the string representation without the "0x" prefix.
+let addressFromStringWithoutPrefix: Address = Address.fromString("436164656E636521")
+
+// Invalid: Provided value is a valid hex string. The function panics.
+let invalidAddressForInvalidHex: Address = Address.fromString("0xZZZ")
```

Comment:
Why is the behavior on invalid input to panic? This is pretty unfriendly to users. The specification in https://github.com/onflow/cadence/issues/2154 says the function should return `nil` on invalid input, which is a friendlier API. Can you change the behavior to have `fromString` return an optional instead of panicking on invalid input?

Reply:
Sure, done.